### PR TITLE
[th2-2212] limited attempts for RabbitMQ connection recovery

### PR DESCRIPTION
**Pull request by andreydrobynin**
Looks like attempts never were reseted and it was waiting for recovery some big time, please take a look on `recoveryDelay` calculation:
```
int recoveryDelay = connectionManagerConfiguration.getMinConnectionRecoveryTimeout()
        + (connectionManagerConfiguration.getMaxRecoveryAttempts() > 1
        ? (connectionManagerConfiguration.getMaxConnectionRecoveryTimeout() - connectionManagerConfiguration.getMinConnectionRecoveryTimeout())
        / (connectionManagerConfiguration.getMaxRecoveryAttempts() - 1)
        * tmpCountTriesToRecovery
        : 0);
```
`setConnectionRecoveryTriggeringCondition()` called only once when it first time faced with connection problems and `tmpCountTriesToRecovery` always was 0.
I moved some code from `setConnectionRecoveryTriggeringCondition()` to `setRecoveryDelayHandler()`.
Related to https://github.com/th2-net/th2-common-j/pull/164.

### CODE REVIEW

**Review by Nikita-Smirnov-Exactpro — CHANGES_REQUESTED** [review]

File: src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java [hunk 1]
```diff
@@ -646,7 +646,11 @@ protected PrometheusConfiguration loadPrometheusConfiguration() {
     }
 
     protected ConnectionManager createRabbitMQConnectionManager() {
-        return new ConnectionManager(getRabbitMqConfiguration(), getConnectionManagerConfiguration(), livenessMonitor::disable);
+        return new ConnectionManager(
+                getRabbitMqConfiguration(),
+                getConnectionManagerConfiguration(),
+                livenessMonitor
```
[the remark this answers is not in the document]
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java]: done

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java [hunk 2]
```diff
@@ -72,11 +74,14 @@
...
             .build());
 
+    private final MetricMonitor livenessMonitor;
     private final HealthMetrics metrics = new HealthMetrics(this);
 
     private final RecoveryListener recoveryListener = new RecoveryListener() {
         @Override
         public void handleRecovery(Recoverable recoverable) {
+            livenessMonitor.enable();
+            LOGGER.debug("Set RabbitMQ liveness to true");
```
[the remark this answers is not in the document]
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java]: done

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java [hunk 3]
```diff
@@ -72,11 +74,14 @@
...
             metrics.getReadinessMonitor().enable();
```
[the remark this answers is not in the document]
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java]: I'm not sure about it. As we discussed now we'll have liveness and readiness metrics in `ConnectionManager` and there are some situations when we should enable both metrics: after `ConnectionManager` creation and after recovery.
I think it's more convinient to call both `enable()/disable()` in one row.

File: src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java [hunk 4]
```diff
@@ -343,7 +350,8 @@ private void waitForConnectionRecovery(ShutdownNotifier notifier, boolean waitFo
     }
 
     private void waitForRecovery(ShutdownNotifier notifier) {
-        LOGGER.warn("Start waiting for connection recovery");
+        metrics.getReadinessMonitor().disable();
+        LOGGER.warn("Start waiting for connection recovery. Set RabbitMQ readiness to false");
         while (isConnectionRecovery(notifier)) {
             try {
                 Thread.sleep(1);
```
[the remark this answers is not in the document]
Comment (andreydrobynin) [reply src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java]: I thought that in this PR I should fix only recovery attempts and nothing more.

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**